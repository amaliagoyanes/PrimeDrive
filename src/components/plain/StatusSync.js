import React from 'react';
import {
    AsyncStorage,
    TouchableOpacity,
    Text,
    View, NetInfo
} from 'react-native';
import firestack from '../../fbconfig.js'
import styles from '../../style/styles';
import {scale, scaleText} from '../../scaling';

export default class StatusSync extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            status: true, number: undefined, client: undefined
        };

        this.handleConnectivityChange = this.handleConnectivityChange.bind(this);
    }

    componentDidUpdate(props, state){
        // if (props.user_uid !== this.props.user_uid)
        //     AsyncStorage.getItem('@PrimeDrive:user_info')
        //         .then(info => {
        //             info = JSON.parse(info);
        //             if (info)
        //                 this.setState({number: info.number});
        //             else
        //                 this.setState({number: ''});
        //         })
        //         .catch(err => console.log(err));
    }

    handleConnectivityChange(isConnected) {
        this.setState({status: isConnected});
    }

    componentDidMount(){
        AsyncStorage.getItem('@PrimeDrive:user_info')
            .then(info => {
                info = JSON.parse(info);
                this.setState({client: this.props.client_name, number: info !== null ? info.number : ''})
            })
            .catch(err => console.log(err));
        NetInfo.isConnected.fetch().then(isConnected => {
            this.setState({status: isConnected})
        });
        NetInfo.isConnected.addEventListener('change', this.handleConnectivityChange);
    }

    componentWillUnmount(){
        NetInfo.isConnected.removeEventListener('change', this.handleConnectivityChange);
    }

    render() {
        let version = '1.0.0';
        let statusText = '';
        // if (this.state.number)
        //     statusText = this.state.number + ' - ';
        if (this.props.userNumber)
            statusText += ' - ';
        if (this.props.client_name)
            statusText += this.props.client_name + ' - ';
        statusText += `V${version}`;

        let styleCircle = {
            borderRadius: scale(7),
            width: scale(14),
            height: scale(14),
            marginRight: 5
        };

        return (
            <TouchableOpacity style={{
                justifyContent: 'center', 
                alignItems: 'center', 
                flexDirection: 'row',
                height: scale(30)
            }} onPress={this.props.forceSync}>
                <View style={{...styleCircle, backgroundColor: this.state.status ? 'green' : 'yellow'}} />
                <Text style={styles.text}>{this.props.userNumber} {statusText}</Text>
            </TouchableOpacity>
        );
    }
}