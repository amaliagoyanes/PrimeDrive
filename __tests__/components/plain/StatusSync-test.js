'use strict';

import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import StatusSync from '../../../src/components/plain/StatusSync';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

StatusSync.prototype.componentDidMount = () => {};

it('renders correctly', () => {
    const tree = renderer.create(
        <StatusSync forceSync={jest.fn} userNumber={1}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with client name', () => {
    const tree = renderer.create(
        <StatusSync forceSync={jest.fn} userNumber={1} client_name="Test client"/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with number', () => {
    const tree = shallow(
        <StatusSync forceSync={jest.fn} userNumber={1}/>
    );
    tree.setState({number: 123});
    expect(tree).toMatchSnapshot();
});

it('renders correctly with number and client name', () => {
    const tree = shallow(
        <StatusSync forceSync={jest.fn} userNumber={1} client_name="Test client"/>
    );
    tree.setState({number: 123});
    expect(tree).toMatchSnapshot();
});
