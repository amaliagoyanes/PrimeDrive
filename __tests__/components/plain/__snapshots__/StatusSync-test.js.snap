// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "green",
        "borderRadius": 7,
        "height": 14,
        "marginRight": 5,
        "width": 14,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Gotham-Book",
        "fontSize": 14,
      }
    }
  >
    1
     
    V1.0.0
  </Text>
</View>
`;

exports[`renders correctly with client name 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "green",
        "borderRadius": 7,
        "height": 14,
        "marginRight": 5,
        "width": 14,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Gotham-Book",
        "fontSize": 14,
      }
    }
  >
    1
     
    Test client - V1.0.0
  </Text>
</View>
`;

exports[`renders correctly with number 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "green",
        "borderRadius": 7,
        "height": 14,
        "marginRight": 5,
        "width": 14,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Gotham-Book",
        "fontSize": 14,
      }
    }
  >
    1
     
    123 - V1.0.0
  </Text>
</TouchableOpacity>
`;

exports[`renders correctly with number and client name 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "green",
        "borderRadius": 7,
        "height": 14,
        "marginRight": 5,
        "width": 14,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Gotham-Book",
        "fontSize": 14,
      }
    }
  >
    1
     
    123 - Test client - V1.0.0
  </Text>
</TouchableOpacity>
`;
